lib/generic/context_runner.rb

```diff
@@ -0,0 +1,103 @@
+module Synthea
```

Comment:
History is shared among all the modules in a stack. The new current_encounter is shared among all the modules in a stack. The only thing that really has to change when you call a submodule is that the current state and "module" have to be saved off in a stack somewhere.  There shouldn't have to be special logic in the `run` function of the Context/ContextRunner that checks if the last state is a CallSubmodule - there should be a new load_submodule function in the Context that gets called from CallSubmodule states. (and there can probably be something for Terminal too) Then the ContextRunner becomes unnecessary. Even though there's a 1-1 mapping between Context-Module right now there doesn't have to be. "Context" really just means the environment where things are happening so it's still the appropriate term if it encapsulates the whole Module-Submodule relationship.

And package is just a folder, we don't need a whole separate class to represent a folder. Also I don't see why modules have to specify what package they are in. The folder name is the package name, why does it have to be specified again within the file itself? And if the "main" module in a package was instead in the "modules" folder instead of the package folder, there doesn't have to be a distinction between a "module" and a "package" at all.